src/main/java/carpettisaddition/mixins/command/player/pertick/ServerPlayerEntityMixin.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of the Carpet TIS Addition project, licensed under the
+ * GNU Lesser General Public License v3.0
+ *
+ * Copyright (C) 2023  Fallen_Breath and contributors
+ *
+ * Carpet TIS Addition is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Carpet TIS Addition is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Carpet TIS Addition.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package carpettisaddition.mixins.command.player.pertick;
+
+import carpettisaddition.helpers.carpet.playerActionEnhanced.IServerPlayerEntity;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.math.Vec2f;
+import net.minecraft.util.math.Vec3d;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(ServerPlayerEntity.class)
+public abstract class ServerPlayerEntityMixin implements IServerPlayerEntity
+{
+	private Vec3d posLastTick;
+	private Vec2f rotLastTick;
+
+	public void pushOldPosRot()
+	{
+		ServerPlayerEntity self = (ServerPlayerEntity)(Object)this;
+		posLastTick = self.getPos();
+		rotLastTick = self.getRotationClient();
+	}
+
+	public void swapOldPosRot(boolean lastTickValues)
```

Comment:
Suggestion: Split this function into two different functions, and don't use bool arg to differentiate the internal logic, because "set" and "restore" are 2 different logics